From de932be8affdabdbb942a19a1a4f04ee7df3bda9 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 25 Aug 2017 18:36:21 +0200
Subject: [PATCH 010/179] arm64: dts: bcm2837: provide for BT-less variant,
 re-enabling uart0 as serial

Raspberry 3 has two UARTs: uart0 is a full-fledged pl011 used for BT
by default, uart1 is a poor-man's low throughput serial device dubbed
as the "mini-uart". Unfortunately, uart1 is a massive pain, whose
clock rate is based on the CPU clock rate: this makes it quite
unstable as a serial console when a CPU frequency governor is enabled.

When BT is useless, we'd rather switch uart0 from BT to the serial
pins, disabling uart1 in the same move, so that we recover a decent
console device (*) with a stable clocking.

bcm2837-rpi-3-b-nobt.dtb is a fixed up blob doing exactly that.

(*) cmdline should mention console=ttyAMA0,<speed>

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm64/boot/dts/broadcom/Makefile                |  1 +
 .../arm64/boot/dts/broadcom/bcm2837-rpi-3-b-nobt.dts | 12 ++++++++++++
 2 files changed, 13 insertions(+)
 create mode 100644 arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b-nobt.dts

diff --git a/arch/arm64/boot/dts/broadcom/Makefile b/arch/arm64/boot/dts/broadcom/Makefile
index cb7de8d99..7f867844b 100644
--- a/arch/arm64/boot/dts/broadcom/Makefile
+++ b/arch/arm64/boot/dts/broadcom/Makefile
@@ -3,6 +3,7 @@ dtb-$(CONFIG_ARCH_BCM2835) += bcm2711-rpi-4-b.dtb \
 			      bcm2837-rpi-3-a-plus.dtb \
 			      bcm2837-rpi-3-b.dtb \
 			      bcm2837-rpi-3-b-plus.dtb \
+			      bcm2837-rpi-3-b-nobt.dtb \
 			      bcm2837-rpi-cm3-io3.dtb
 
 subdir-y	+= northstar2
diff --git a/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b-nobt.dts b/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b-nobt.dts
new file mode 100644
index 000000000..43f9d0f66
--- /dev/null
+++ b/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b-nobt.dts
@@ -0,0 +1,12 @@
+/dts-v1/;
+#include "bcm2837-rpi-3-b.dts"
+
+&uart0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_gpio32>;
+};
+
+&uart1 {
+	status = "disabled";
+};
-- 
2.38.1

